docs/DeveloperGuide.md

```diff
@@ -231,32 +264,67 @@ The following sequence diagram summarizes what happens when the user inputs an a
 
 ![AddGroupSequenceDiagram](images/AddGroupSequenceDiagram.png)
 
-### Add Allocation feature
+### Add Allocation : `addalloc`
+
+The `addalloc` feature allows users to allocate a student into a group.
+
+#### How the `AddAllocCommand` works
 
-The `addalloc` feature allocates an existing student into a group.
+Step 1. The user specifies the group name, the name or ID of the student to be allocated into the group.
```

Comment:
Don't use "Step 1.".
Instead just use the standard numbered list (i.e. 1.)

"Step 1" implies that it's a step that we actively do. In actual fact, it's just the computer doing all the work.